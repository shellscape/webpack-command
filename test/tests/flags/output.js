const { apply, build, test, validate } = require('../../util');

const fixtures = [
  'output-chunk-filename',
  'output-filename',
  'output-jsonp-function',
  'output-library-target',
  'output-library',
  'output-path',
  'output-pathinfo',
  'output-public-path',
  'output-source-map-filename',
  'output'
];

test('--output-*', module, () => {
  for (const name of fixtures) {
    const fixture = `output/${name}`;
    const opts = { fixture };

    let config;

    it(`${name} should validate`, () => {
      expect(validate(opts)).toEqual(true);
    });

    it(`${name} should apply`, () => {
      config = apply(opts);

      expect(config).toMatchSnapshot();
    });

    it(`${name} should build`, () =>
      build(config).then((result) => {
        expect(result).toMatchSnapshot();
        // expect(crcDist()).toMatchSnapshot();
      }));
  }
});
