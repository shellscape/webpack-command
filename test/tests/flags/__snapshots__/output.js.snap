// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --output-* > output should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "filename": "bundle.js",
    "path": "<PROJECT_ROOT>/dist/.output",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output should build #0 1`] = `
"Version: webpack 4.39.2
    Asset      Size  Chunks             Chunk Names
bundle.js  4.47 KiB    main  [emitted]  main
Entrypoint main = bundle.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-chunk-filename should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "chunkFilename": "[id].output-chunk.js",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-chunk-filename should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.47 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-filename should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "filename": "output-[name].js",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-filename should build #0 1`] = `
"Version: webpack 4.39.2
         Asset      Size  Chunks             Chunk Names
output-main.js  4.47 KiB    main  [emitted]  main
Entrypoint main = output-main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-jsonp-function should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "jsonpFunction": "output-func",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-jsonp-function should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.47 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-library should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "library": "__output__",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-library should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.51 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-library-target should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "libraryTarget": "var",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-library-target should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.47 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-path should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/dist/.output",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-path should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.47 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-pathinfo should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "pathinfo": true,
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-pathinfo should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.47 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-public-path should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "publicPath": "/output-path/",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-public-path should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.48 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;

exports[`Flags > --output-* > output-source-map-filename should apply #0 1`] = `
Object {
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/output/output-entry.js",
  "mode": "development",
  "output": Object {
    "sourceMapFilename": "output-[name].map",
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --output-* > output-source-map-filename should build #0 1`] = `
"Version: webpack 4.39.2
          Asset      Size  Chunks             Chunk Names
        main.js  4.35 KiB    main  [emitted]  main
output-main.map  3.86 KiB    main  [emitted]  main
Entrypoint main = main.js output-main.map
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/flags/output/output-entry.js] 79 bytes {main} [built]"
`;
